src/deps-index.ts

```diff
@@ -81,28 +29,176 @@ function _getHtmlDependencies(
  */
 export async function buildDepsIndex(
     entrypoints: ResolvedUrl[], analyzer: Analyzer): Promise<DepsIndex> {
-  const depsIndex = {
-    entrypointToDeps: new Map<ResolvedUrl, Set<ResolvedUrl>>()
-  };
+  const depsIndex = new Map<ResolvedUrl, Set<ResolvedUrl>>();
   const analysis = await analyzer.analyze(entrypoints);
   const allEntrypoints = new Set<ResolvedUrl>(entrypoints);
+  const inlineDocuments = new Map<ResolvedUrl, Document>();
 
   // Note: the following iteration takes place over a Set which may be added
   // to from within the loop.
   for (const entrypoint of allEntrypoints) {
     try {
-      const document = getAnalysisDocument(analysis, entrypoint);
-      const deps = getHtmlDependencies(document);
-      depsIndex.entrypointToDeps.set(
-          entrypoint, new Set([entrypoint, ...deps.eagerDeps]));
+      const document = inlineDocuments.has(entrypoint) ?
+          inlineDocuments.get(entrypoint)! :
+          getAnalysisDocument(analysis, entrypoint);
+      const deps = getDependencies(analyzer, document);
+      depsIndex.set(entrypoint, new Set([
+                      ...(document.isInline ? [] : [document.url]),
+                      ...deps.eagerDeps
+                    ]));
+
       // Add lazy imports to the set of all entrypoints, which supports
       // recursive
       for (const dep of deps.lazyImports) {
         allEntrypoints.add(dep);
       }
+
+      // Treat top-level script imports as entrypoints by creating "sub-bundle"
+      // URLs to enable the inline document to be processed as an entrypoint
+      // document on a subsequent iteration of the outer loop over all
+      // entrypoints.
+      for (const [id, imported] of deps.moduleScriptImports) {
+        const subBundleUrl = getSubBundleUrl(document.url, id);
+        allEntrypoints.add(subBundleUrl);
+        inlineDocuments.set(subBundleUrl, imported);
+      }
     } catch (e) {
       console.warn(e.message);
     }
   }
+
   return depsIndex;
 }
+
+/**
+ * Constructs a ResolvedUrl to identify a sub bundle, which is a concatenation
+ * of the super bundle or containing file's URL and an id for the sub-bundle.
+ */
+export function getSubBundleUrl(
+    superBundleUrl: ResolvedUrl, id: string): ResolvedUrl {
+  return `${superBundleUrl}>${id}` as ResolvedUrl;
+}
+
+/**
+ * Strips the sub-bundle id off the end of a URL to return the super bundle or
```

Comment:
Added examples to this comment in commit 0bac76fe57ab89a414ff2a840a332059c81cf0e7.

Like this:
```
/**
 * Strips the sub-bundle id off the end of a URL to return the super bundle or
 * containing file's URL.  If there is no sub-bundle id on the provided URL, the
 * result is essentially a NOOP, since nothing will have been stripped.
 *
 * Sub-Bundle URL for an inline ES6 module document:
 *     file:///my-project/src/page.html>inline#1>es6-module
 *
 * Super-bundle URL for the containing HTML document:
 *     file:///my-project/src/page.html
 */
```